vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -190,6 +189,26 @@ ReactInstanceWin::ReactInstanceWin(
           this,
           options.Properties,
           winrt::make<implementation::ReactNotificationService>(options.Notifications))} {
+  m_whenLoaded.AsFuture()
+      .Then<Mso::Executors::Inline>(
+          [onLoaded = m_options.OnInstanceLoaded, reactContext = m_reactContext](Mso::Maybe<void> &&value) noexcept {
+            if (onLoaded) {
+              onLoaded.Get()->Invoke(reactContext, value.IsError() ? value.TakeError() : Mso::ErrorCode());
+            }
+          })
+      .Then(Queue(), [whenLoaded = std::move(whenLoaded)](Mso::Maybe<void> &&value) noexcept {
+        whenLoaded.SetValue(std::move(value));
+      });
+  m_whenDestroyedResult =
+      m_whenDestroyed.AsFuture().Then<Mso::Executors::Inline>([whenLoadFailed = m_whenLoaded,
+                                                               onDestroyed = m_options.OnInstanceDestroyed,
+                                                               reactContext = m_reactContext]() noexcept {
+        whenLoadFailed.TryCancel();
+        if (onDestroyed) {
+          onDestroyed.Get()->Invoke(reactContext);
```

Comment:
The old code had it's null check onthe result of Get() is that no longer needed?

Reply:
It is the same: checking for null has the same effect as the explicit cast to bool.
BTW, we could call the callback as `onDestroyed(reactContext);`. The reason we use longer syntax is avoid one extra entry in the call stack. The `onDestroyed(reactContext);` internally calls `Invoke(reactContext)`. Thus, the longer syntax removes the `onDestroyed(reactContext);` from the call stack.